File: app/src/main/java/sgtmelon/scriptum/office/intf/ItemIntf.java
package sgtmelon.scriptum.office.intf;

import android.view.View;

import androidx.annotation.NonNull;

public interface ItemIntf {

    interface ClickListener {
        void onItemClick(@NonNull View view, int p);
    }

    interface LongClickListener {
        boolean onItemLongClick(@NonNull View view, int p);
    }

    interface DragListener {
        void setDrag(boolean drag);
    }

    interface RollWatcher {
        void afterRollChanged(int p, @NonNull String text);
    }

}

File: app/src/main/java/sgtmelon/scriptum/office/intf/ItemIntf.kt
package sgtmelon.scriptum.office.intf

import android.view.View

interface ItemIntf {

    interface ClickListener {
        fun onItemClick(view: View, p: Int)
    }

    interface LongClickListener {
        fun onItemLongClick(view: View, p: Int)
    }

    interface DragListener {
        fun setDrag(drag: Boolean)
    }

    interface RollWatcher {
        fun afterRollChanged(p: Int, text: String)
    }

}